File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int t : tangerine) {
            if (map.containsKey(t)) {
                map.put(t, map.get(t) + 1);
            } else {
                map.put(t, 1);
            }
        }
        
        List<Map.Entry<Integer, Integer>> list = new ArrayList<>(map.entrySet());
        Collections.sort(list, (a, b) -> b.getValue().compareTo(a.getValue()));
        
        int answer = 0;
        for (Map.Entry entry : list) {
            int key = (int) entry.getKey();
            int value = (int) entry.getValue();

            answer++;
            k -= value;
            if (k <= 0) break;
        }
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    counter = Counter(tangerine)
    
    answer = 0
    for key, value in sorted(counter.items(), key=lambda x : -x[1]):
        answer += 1
        k -= value
        if k <= 0:
            break
    return answer